src/Confluent.Kafka/IAdminClient.cs

```diff
@@ -529,6 +529,17 @@ public interface IAdminClient : IClient
         ///     A Task with an empty result when successful.
         /// </returns>
         Task AlterUserScramCredentialsAsync(IEnumerable<UserScramCredentialAlteration> alterations, AlterUserScramCredentialsOptions options = null);
+        /// <summary>
+        ///     ListOffsets given the topic_partitions with the OffsetSpec
+        /// </summary>
+        /// <param name="requests">
+        ///     The mapping of TopicPartitions to OffsetSpec.
+        /// </param>
+        /// <param name="options">
+        ///     The options to use while List Offsets.
+        /// </param>
+        Task<ListOffsetsResult> ListOffsetsAsync(
+            Dictionary<TopicPartition,OffsetSpec> requests, ListOffsetsOptions options = null);
```

Comment:
I think we can create a new class rather than keeping a dictionary here something like `TopicPartitionOffsetSpec`